### PR TITLE
Authentication Part 2

### PR DESCRIPTION
**Pull request by SonaliChavan95**

### CODE REVIEW

**Review by neilvarunnaidu — APPROVED** [review]

File: data/users.js [hunk 1]
```diff
@@ -1,62 +1,90 @@
...
+
+    let user = User.create({
+      firstname: userInfo.firstname,
+      lastname: userInfo.lastname,
+      email: userInfo.email,
+      hashedPassword: hash,
+      gender: userInfo.gender,
+      role_id: userInfo.role_id,
+    });
+    console.log(user);
```
Comment (abhinav098) [inline data/users.js:55]: please remove this console.log

File: helpers/auth.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+function loginRequired(req, res, next) {
```
Comment (abhinav098) [inline helpers/auth.js:1]: could you please rename this file to authMiddlewares ?

Or maybe move it to middlewares/ directory.

File: routes/auth.js [hunk 3]
```diff
@@ -1,43 +1,22 @@
 const express = require('express');
 const router = express.Router();
-const bcrypt = require('bcryptjs');
 const data = require('../data');
 const userData = data.users;
+const { Role } = require('../models');
+const authHelpers = require('../helpers/auth');
```
Comment (abhinav098) [inline routes/auth.js:6]: please rename this to `authMiddlewares`

File: routes/auth.js [hunk 4]
```diff
@@ -46,24 +25,35 @@ router.post('/login', async (req, res) => {
...
+    let userInfo = req.body;
+    console.log(userInfo);
+    userData.validateUserInfo(userInfo);
+    let user = await userData.create(userInfo);
+    if (user) {
+      req.session.user = user;
+    }
+    res.redirect('/');
+  } catch (error) {
+    res.status(400).json({ error: error });
```
Comment (abhinav098) [inline routes/auth.js:55]: 422 is more relevant here.

```
The HyperText Transfer Protocol (HTTP) 422 Unprocessable Entity response status code 
indicates that the server understands the content type of the request entity, and the 
syntax of the request entity is correct, but it was unable to process the 
contained instructions.
```

**Review by abhinav098 — CHANGES_REQUESTED** [review]

File: helpers/auth.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+function donorRequired(res, req, next) {
+  if (!req.session.user) return res.redirect('/auth/login');
+
+  return next();
+}
+
+function recipientRequired(res, req, next) {
+  return next();
+}
```
Comment (abhinav098) [inline helpers/auth.js:22]: Did you miss something here?

**Review by abhinav098 — CHANGES_REQUESTED** [review]
Awesome work on auth. Minor comments added.

File: routes/auth.js (hunk 4 above)
Comment (SonaliChavan95) [reply routes/auth.js:55]: @abhinav098 I will read more on this and will update later accordingly.

File: helpers/auth.js (hunk 5 above)
Comment (SonaliChavan95) [reply helpers/auth.js:22]: I will be updating this code in the next PR.

**Review by abhinav098 — APPROVED** [review]

**Review by abhinav098 — APPROVED** [review]

**Pull request merged**